app-backend/backsplash-core/src/main/scala/com/rasterfoundry/backsplash/BacksplashImage.scala

```diff
@@ -72,15 +82,35 @@ final case class BacksplashGeotiff(
           }
         }
       } else {
-        logger.debug(s"Using GeoTiffRasterSource: ${uri}")
-        val rasterSource = new GeoTiffRasterSource(uri)
-        IO {
-          metadata.noDataValue match {
-            case Some(nd) =>
-              rasterSource.interpretAs(DoubleUserDefinedNoDataCellType(nd))
-            case _ =>
-              rasterSource
+        for {
+          infoOption <- child.childSpan("getSceneGeoTiffInfo") use { _ =>
+            SceneDao.getSceneGeoTiffInfo(imageId).transact(xa)
+          }
+          info <- child.childSpan("getGeotiffInfoFromSource") use { _ =>
+            infoOption match {
+              case Some(i) => IO.pure(i)
+              case _ => {
+                for {
+                  sourceInfo <- IO(
+                    BacksplashGeotiffReader.getBacksplashGeotiffInfo(uri))
+                  _ <- SceneDao
```

Comment:
ugh, it finally happened :disappointed: :disappointed: :disappointed: :disappointed:

Reply:
The `database` code finally making its way into `backsplash-image` makes me think there's an abstracted version of this work that probably belongs in gt-server, but I'm not sure what the issue for that would look like

Reply:
What are you thinking can be abstracted?

There's probably some big picture stuff that I missed in this refactor because this set of changes started as incremental until there was a (rather large) qualitative shift in where things ended up.